src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -809,6 +816,25 @@ private void WriteDynamicComplexProperty(object propertyValue, IEdmTypeReference
             }
         }
 
+        private void WriteReferencedNavigationProperties(ISet<IEdmNavigationProperty> referencedPropertiesToExpand,
+            ResourceContext resourceContext, ODataWriter writer)
+        {
+            Contract.Assert(referencedPropertiesToExpand != null);
```

Comment:
This does not seem right. This can be null if there no referenced navigation properties.

Reply:
Thanks. But it's right, because the SelectExpandNode always creates a HashSet for it.